schema/settings.json

```diff
@@ -46,6 +46,12 @@
       "title": "History of Cell Execution Time",
       "description": "Show the historical N previous execution times since the notebook was opened as a tooltip (0 to disabled)",
       "default": 5
+    },
+    "dateFormat": {
+      "type": "string",
+      "title": "Date format",
+      "description": "Show the formatted date string in the given format. For example \"yyy-MM-dd HH:mm:ss\" results in \"2023-02-28 14:25:57 \", \"dd/MM/YYY HH:mm:ss (O)\" results in \"28/02/2023 14:25:57 (GMT +3)\".Find out more on unicode date field symbols on this link [https://www.unicode.org/reports/tr35/tr35-dates.html#Date_Field_Symbol_Table]",
+      "default": "yyy-MM-dd HH:mm:ss"
```

Comment:
This way users who use the settings form will not enter invalid format in the first place.

```suggestion
      "default": "yyy-MM-dd HH:mm:ss",
      "pattern": "[yGuqQMLwdeEcabBhHkKmsSzOxX\\W].*"
```

Reply:
That is a good suggestion. Thank you.  
Could you point me to where 'pattern' is mentioned in the developer docs?

Reply:
How does the pattern work? is it added/injected to the input variable as an attribute

Reply:
Settings use JSON schema, the `pattern` is documented here: https://json-schema.org/understanding-json-schema/reference/regular_expressions.html
In JupyterLab the settings are validated using [ajv](https://ajv.js.org/) validator. It works reliably in the "Advanced Settings Editor" preventing saving until it is corrected:

![Screenshot from 2023-09-09 10-42-41](https://github.com/deshaw/jupyterlab-execute-time/assets/5832902/4c5f7b83-6b22-4f2f-9444-33d07ec0ea9a)

And it usually works in the settings UI form:

![Screenshot from 2023-09-09 10-44-56](https://github.com/deshaw/jupyterlab-execute-time/assets/5832902/070237c2-8eac-45f9-8c45-9820448d188c)

But there is some room for improvement, because some strings appear not to be picked up due to partial match (fortunately with the dialog approach user is still warned, even if they used another way to set settings, e.g. [`overrides.json` file](https://jupyterlab.readthedocs.io/en/stable/user/directories.html#overrides-json))

![Screenshot from 2023-09-09 10-44-43](https://github.com/deshaw/jupyterlab-execute-time/assets/5832902/cce3c1f3-4f0d-4db2-99b6-634dfc45881b)

Reply:
Thank you for the elaborate explanation.

I noticed the validation error does not appear when inputting some string. The show error dialog is a good caveat.